src/silicone/utils.py

```diff
@@ -344,54 +352,46 @@ def convert_units_to_MtCO2_equiv(df, use_AR4_data=False):
         The input data with units converted.
     """
     # Check things need converting
-    if all(y[0:6] == "Mt CO2" for y in df.variables(True)["unit"]):
+    convert_to_str = "Mt CO2-equiv/yr"
+    if (df["unit"] == convert_to_str).all():
         return df
-    if use_AR4_data:
-        file = "../../Input/GWP100_unit_conversion_AR4.csv"
-    else:
-        file = "../../Input/GWP100_unit_conversion_AR5.csv"
-    conversion_factors = pd.read_csv(
-        os.path.join(os.path.dirname(__file__), file), sep=";", header=3
-    )
-    # This string is found at the start of all correct units:
-    convert_to_str = "Mt CO2"
+
+    convert_to_str_clean = "Mt CO2/yr"
+
+    context = "AR4GWP100" if use_ar4_data else "AR5GWP100"
+
     to_convert_df = df.copy()
     to_convert_var = to_convert_df.filter(
-        unit=convert_to_str + "*", keep=False
+        unit=convert_to_str, keep=False
     ).variables(True)
     to_convert_units = to_convert_var["unit"]
-    not_found = [
-        y
-        for y in to_convert_units.map(
-            lambda x: x.split(" ")[-1][:-3].replace("-equiv", "")
-        ).values
-        if y not in conversion_factors["Gas"].values
-    ]
+    to_convert_units_clean = {
+        unit: unit.replace("-equiv", "").replace("equiv", "")
```

Comment:
Is there a case with "equiv", no hyphen and no space? I feel like there'd normally be a space there if no hyphen, but I wasn't aware of cases in the data missing the hyphen.

Reply:
not sure, I'll add a test just to make sure we cover all those bases anyway

Reply:
solved by 6f7fe3df